en/debug/gdb_debugging.md

```diff
@@ -70,12 +70,32 @@ The numbers in the graph will all be zero, because they are in MB. Just look at
 See the [gperftools docs](https://htmlpreview.github.io/?https://github.com/gperftools/gperftools/blob/master/docs/heapprofile.html) for more information.
 
 
-## Debugging Hard Faults in NuttX
+## Hard Fault Debugging
 
-A hard fault is a state when the operating system detects that it has no valid instructions to execute. This is typically the case when key areas in RAM have been corrupted. A typical scenario is when incorrect memory access smashed the stack and the processor sees that the address in memory is not a valid address for the microprocessors's RAM.
+A hard fault is a state when the operating system detects that it has no valid instructions to execute.
```

Comment:
@hamishwillee  - please update this to say:

A hard fault is a state when CPU executes an invalid instruction or accesses an invalid memory address. 
This is typically the case when key areas in RAM have been corrupted.

Video

The following video demonstrates hardfault debugging on PX4 using Eclipse and a JTAG debugger. It was presented at the PX4 Developer Conference 2019.
{% youtube %} https://www.youtube.com/watch?v=KZkAM_PVOi0 {% endyoutube %}

Debugging Hard Faults in NuttX

A typical scenario that can cause a hard fault is when the processor overwrites the stack and then processor returns to an invalid address from the stack.  This may be caused by a bug in code were wild pointer corrupts the stack, or another task overwrites this tasks stack.

Reply:
Thanks, that makes a lot more sense. Fixed in https://github.com/PX4/Devguide/pull/941 (with minor subedit).